networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/StatusMessageFactory.java

```diff
@@ -111,14 +116,33 @@ public Optional<StatusMessage> createStatusMessage(final StatusMessageSchema<?>
 
   @Override
   public void onSlot(final UInt64 slot) {
-    if (spec.computeStartSlotAtEpoch(combinedChainDataClient.getCurrentEpoch()).equals(slot)) {
+    final boolean isStartOfEpoch =
+        spec.computeStartSlotAtEpoch(combinedChainDataClient.getCurrentEpoch()).equals(slot);
+    if (isStartOfEpoch && supportsEarliestAvailableSlot) {
       updateEarliestAvailableSlot();
     }
   }
 
   private void updateEarliestAvailableSlot() {
-    combinedChainDataClient
-        .getEarliestAvailableBlockSlot()
+    final SafeFuture<Optional<UInt64>> earliestAvailableBlockSlotFuture =
+        combinedChainDataClient.getEarliestAvailableBlockSlot();
+    final SafeFuture<Optional<UInt64>> earliestDataColumnSidecarSlotFuture =
+        combinedChainDataClient.getEarliestDataColumnSidecarSlot();
+
+    final BiFunction<Optional<UInt64>, Optional<UInt64>, Optional<UInt64>>
+        combineEarliestSlotFunction =
+            (blockEarliestAvailableSlot, dataColumnEarliestAvailableSlot) -> {
+              if (blockEarliestAvailableSlot.isPresent()
+                  && dataColumnEarliestAvailableSlot.isPresent()) {
+                return Optional.ofNullable(
```

Comment:
I don't feel this could be null (if so, we NPE anyway during calculation), so we can use `Optional.of`